src/core/Badge.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { PropsWithChildren, useMemo } from "react";
+import { IconName, IconSize } from "./Icon/types";
+import Icon from "./Icon";
+import cn from "./utils/cn";
+import { ColorClassColorGroups } from "./styles/colors/types";
+
+/**
+ * Props for the Badge component.
+ */
+interface BadgeProps {
+  /**
+   * The size of the badge. Can be one of "xs", "sm", "md", or "lg".
+   */
+  size?: "xs" | "sm" | "md" | "lg";
+
+  /**
+   * The color of the badge. Can be a value from ColorClassColorGroups or "red".
+   */
+  color?: ColorClassColorGroups | "red";
+
+  /**
+   * The name of the icon to be displayed before the children in the badge.
+   */
+  iconBefore?: IconName;
+
+  /**
+   * The name of the icon to be displayed after the children in the badge.
+   */
+  iconAfter?: IconName;
+
+  /**
+   * Additional CSS class names to apply to the badge.
+   */
+  className?: string;
+
+  /**
+   * Whether the badge is disabled. Defaults to false.
+   */
+  disabled?: boolean;
+
+  /**
+   * Whether the badge is focusable. Defaults to false.
+   */
+  focusable?: boolean;
+
+  /**
+   * Whether the badge is hoverable. Defaults to false.
+   */
+  hoverable?: boolean;
+
+  /**
+   * The size of the icons in the badge. Defaults to 12px.
+   */
+  iconSize?: IconSize;
+
+  /**
+   * Accessible label for the badge when interactive
+   */
+  ariaLabel?: string;
+}
+
+const Badge: React.FC<PropsWithChildren<BadgeProps>> = ({
+  size = "md",
+  color = "neutral",
+  iconBefore,
+  iconAfter,
+  className,
+  children,
+  disabled = false,
+  focusable = false,
+  hoverable = false,
+  iconSize = "12px",
+  ariaLabel,
+}) => {
+  const sizeClass = useMemo(() => {
+    switch (size) {
+      case "xs":
+        return "px-8 py-0 text-[10px] leading-tight";
+      case "sm":
+        return "px-8 py-2 text-[10px] leading-tight";
+      case "md":
+        return "px-[10px] py-2 text-[11px] leading-normal";
+      case "lg":
+        return "px-12 py-[3px] text-[12px] leading-normal";
+    }
+  }, [size]);
+
+  const childClass = useMemo(() => {
+    switch (size) {
+      case "xs":
+      case "sm":
+        return "leading-[18px]";
+      case "md":
+      case "lg":
+        return "leading-[20px]";
+    }
+  }, [size]);
+
+  const colorClass = useMemo(() => {
+    switch (color) {
+      case "neutral":
+        return "text-neutral-800 dark:text-neutral-500";
+      case "violet":
+        return "text-violet-400";
+      case "orange":
+        return "text-orange-600";
+      case "yellow":
+        return "text-yellow-600";
+      case "green":
+        return "text-green-600";
+      case "blue":
+        return "text-blue-600";
+      case "pink":
+        return "text-pink-600";
+      case "red":
+        return "text-orange-700";
+    }
+  }, [color]);
+
+  return (
+    <div
+      className={cn(
+        "inline-flex bg-neutral-100 dark:bg-neutral-1200 rounded-2xl gap-4 items-center focus-base transition-colors select-none",
```

Comment:
One comment from Jamie from the badge was the letter spacing it says 4%. for the  `xs` I have added  `tracking-[0.025rem]`

Reply:
You're right about adding letter spacing - I think we need a value of `0.04em` for 4% as that's 4% of the current font size, which then scales for all the sizes. `0.025rem` works nicely for `xs` as that's 2.5% of 1rem (16px) which is the same as to 4% of 10px - both are 0.4.

I've seen 4% around a lot in designs, I'll add it as a token that we can refer to later in other things.